ca/config.go

```diff
@@ -129,10 +129,26 @@ func (s *SecurityConfig) UpdateRootCA(cert, key []byte, certExpiry time.Duration
 	defer s.mu.Unlock()
 
 	rootCA, err := NewRootCA(cert, key, certExpiry)
-	if err == nil {
-		s.rootCA = &rootCA
+	if err != nil {
+		return err
+	}
+
+	// the RootCA pool should validate against the TLS certificate in the credentials
+	if s.ClientTLSCreds != nil {
+		previousConfig := s.ClientTLSCreds.Config()
+		// can't error because we know for sure the root pool is valid
+		newConfig, _ := NewClientTLSConfig(&previousConfig.Certificates[0], rootCA.Pool, previousConfig.ServerName)
```

Comment:
@aaronlehmann Good point - I guess the lock needs to be on the mutable TLS object.